test/cql-pytest/test_describe.py

```diff
@@ -552,6 +552,28 @@ def test_whitespaces_in_table_options(cql, test_keyspace, scylla_only):
         desc = "\n".join([d.create_statement for d in cql.execute(f"DESC TABLE {tbl}")])
         assert re.search(regex, desc) == None
 
+# In order to restore schema from `DESC SCHEMA`
+# UDTs have to be sorted topologically because they can depent on each other
+# The test doesn't check validity of UDT description, 
+# but only the order in `DESC KEYSPACE ks`(it's equivalent to part of `DESC SCHEMA`)
+def test_udts_order(scylla_only, cql, test_keyspace):
```

Comment:
Would it be possible to write a boost unit test that would check multiple orders? Perhaps we could generate those orders randomly with some fixed seed which would easily let us cover a lot of cases.

Reply:
It doesn't need to be a boost test. A cql-pytest can easily loop and try multiple variants of the test, or even use "parametrize" if you prefer.

Reply:
And I do agree that I would have tested more graphs. Your code is fairly complex with a lot of different functions and weird cases, and I'm not convinced that this one example covers all of them.

Reply:
My rationale behind recommending a boost test was that it will have better performance and we will be able to test more graphs there. We could generate a lot of random graphs, increasing the confidence in the algorithm by a lot.

Another thing is that with `cql-pytest` we won't be able to test the case where we have a cycle. You can't introduce a cyclical dependency on UDTs via CQL, we check that during `CREATE TYPE` as far as I can tell. It might be possible to create a set of UDTs with cyclical dependencies in a boost test.

Reply:
Sorting won't work on a graph with cycles. I'm not handling this in the code because there cannot be cycles in UDTs (even by adding new fields with `ALTER TYPE`)

Reply:
> I'm not handling this in the code because there cannot be cycles in UDTs

I know. However, we should handle this case anyway as a sanity check - it's very easy to do (if there is a cycle, the algorithm won't put all types to the vector). I was just wondering whether it would be possible to cover this sanity check in the tests, but perhaps it's not worth the effort.